spaceMissionManager-service/src/main/java/cz/muni/fi/services/impl/CraftComponentServiceImpl.java

```diff
@@ -0,0 +1,49 @@
+package cz.muni.fi.services.impl;
+
+import cz.muni.fi.dao.CraftComponentDao;
+import cz.muni.fi.entity.CraftComponent;
+import cz.muni.fi.services.CraftComponentService;
+import org.springframework.dao.DataAccessException;
+import org.springframework.stereotype.Service;
+
+import javax.inject.Inject;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Implementation of the {@link CraftComponentService}. This class is part of the
+ * service module of the application that provides the implementation of the
+ * business logic (main logic of the application).
+ *
+ * @author Vojtech Bruza
+ */
+@Service
+public class CraftComponentServiceImpl implements CraftComponentService {
+    @Inject
+    private CraftComponentDao craftComponentDao;
+
+    @Override
```

Comment:
DataAccessException - I suggest using try catch and throwing a DataAccessException in catch block
add InvalidArgumentException if given argument is null

Reply:
I also don't know which exceptions I can expect...I guess that we do not handle exceptions on lower levels that properly. Can I use something like "catch (Exception e);"?

Reply:
Create a new Exception which will extend DataAccessException.

You can use catch (Throwable t)